### PR TITLE
BUGFIX: FileTypeIconViewHelper should work with new $asset argument

### LINKED ISSUE #2186: BUG: FileTypeIconViewHelper: if "asset" is used, $asset is resetted to null
### Description
Package **Neos.Media**
Neos.Media/Classes/ViewHelpers/FileTypeIconViewHelper.php
**FileTypeIconViewHelper** since version 4.1
- line 78
- as **$file** is DEPRECATED now ...
- **Bug**: if "**asset**" is used in consequence, **it will be resetted to null ... !** (as $file=null)
### Steps to Reproduce

1. set $asset instead of $file
2. '**$asset = $file**' ... will reset it to null ... line 78
3. ==> **EXCEPTION**: "You must either specify "**asset**" or "filename" ..."

#### Expected behavior

- $asset must keep its value (if set)
- so, to prevent finally this issue:
 **if (!is_null($file)** {
     $asset = $file
}
...
#### Actual behavior

- $asset is resetted to null

### Affected Versions

**Neos.Media  version 4.1**
since commit 4b04a54b7886750f1b4f2af6d06b1cbc917d7edf   18. April 2018

### PR DESCRIPTION
**Pull request by ChKoeppel**
If the newly added `$asset` argument is used, it's ignored and overridden to the deprecated `$file` argument.

Resolves #2186

### CODE REVIEW

**Comment by ChKoeppel** [comment]
Hi
I thought to be able for such a small selective code-fix ...
Apparently not ... I don't have an idea what's really wrong ?  (Type "" not found ??)
Thanks for hint ...

**Comment by daniellienert** [comment]
Hey @ChKoeppel,

thanks for your first contribution to this repository! 

The failed test is not your fault - have a look at the current build status: https://travis-ci.org/neos/neos-development-collection/branches - 4.1 is also broken without your change.

Cheers,
Daniel

File: Neos.Media/Classes/ViewHelpers/FileTypeIconViewHelper.php [hunk 1]
```diff
@@ -75,7 +75,9 @@ public function initializeArguments()
      */
     public function render(AssetInterface $file = null, AssetInterface $asset = null, string $filename = null, $width = null, $height = null)
     {
-        $asset = $file;
+        if (!is_null($file)) {
```
Comment (daniellienert) [inline Neos.Media/Classes/ViewHelpers/FileTypeIconViewHelper.php]: Mhm with that, if you give both $asset and $file, the deprecated file would take precedence.

Nitpick using `$file !== null` is more common in the Neos code.

I would recomend:

```
if($asset === null) {
  $asset = $file;
}
```

Comment (ChKoeppel) [reply Neos.Media/Classes/ViewHelpers/FileTypeIconViewHelper.php]: Thank you
 - shall I adapt the code to your solution **HERE** ... 
(another commit, no ?)
**OR** create a new PullRequest ? (squash)

**Comment by daniellienert** [comment]
Sure, just amend your commit or add another and we’ll squash it while merging.

**Comment by aertmann** [comment]
Should actually go into 4.0 as that's the lowest branch the bug exists in https://github.com/neos/neos-development-collection/blob/4.0/Neos.Media/Classes/ViewHelpers/FileTypeIconViewHelper.php#L78

**Comment by ChKoeppel** [comment]
Really **SORRY** for all these inconvenience ...
I guess now, I have done **another** real mistake ... ??!

Should I have closed and reopen a new one ?!
Can you help me ?

**Comment by gerhard-boden** [comment]
You'll have to reopen with the correct target branch (just reference the new PR from here and close this one). You can't change the target to a lower branch within a PR without this happening afaik, only higher is possible.

**Comment by aertmann** [comment]
@ChKoeppel Really don't be sorry, it's a bit of a complicated process when having to maintain multiple versions simultaneously. Happy to guide you through it. You haven't made any mistakes per say, we just need to make sure we target the lowest version with the bug so it can be fixed in all versions it's present in.

You have two options, either you can reset your existing branch to be based on 4.0 and then apply the change again and force push it. Alternatively you can create a new branch and open a new pull request. Either is fine, depends on your Git skill level.

**Comment by aertmann** [comment]
If you wanna keep the existing PR and just update your branch, then you can change the base by clicking "Edit" next to the title.

**Comment by aertmann** [comment]
Nvm you already did, then you just need to update your branch to be mergeable with 4.0 with your change applied on top.

**Comment by gerhard-boden** [comment]
@aertmann I saw he is using 4.1 within his fork as base so while he could reset his 4.1 branch to the latest 4.0 commit (and put his commits on top), I think from this point it would be less complicated to just reopen? Maybe on a new bugfix branch based on 4.0?

@ChKoeppel As Aske already said, no problem at all, we're all very happy to guide you trough the process which can be a little complicated at the beginning. Just ask anything that is unclear for you to proceed.

**Comment by ChKoeppel** [comment]
... just want to be sure what I do now, - I need some minutes of reflection before trying ...  
Thanks anyway for your hints.

**Pull request closed without merging**